solver/src/driver.rs

```diff
@@ -410,19 +410,26 @@ impl Driver {
             let name = solver.name();
 
             let mut settlements = match settlements {
-                Ok(settlement) => {
+                Ok(mut settlement) => {
+                    solver_settlements::filter_empty_settlements(&mut settlement);
+                    if settlement.is_empty() {
+                        self.metrics.solver_run_empty(name);
+                        continue;
+                    }
+
                     self.metrics.solver_run_succeeded(name);
                     settlement
                 }
                 Err(err) => {
-                    self.metrics.solver_run_failed(name);
+                    match err {
+                        SolverRunError::Timeout => self.metrics.solver_run_timeout(name),
+                        SolverRunError::Solving(_) => self.metrics.solver_run_failed(name),
```

Comment:
I guess there are different methods in the metrics so we don't have to give them the `SolverError` type. If this is actually an issue, we could cast the error type to string and pass it into a single metrics method.

Reply:
This was the inner anyhow Error could at least be separated without a match.

Reply:
Labels on prometheus metrics should be used sparingly. We shouldn't use the error string as a label.

Reply:
ok then.